src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const dateValues = date.split(separator);
+  const dateParts = {};
+
+  for (let i = 0; i < dateValues.length; i++) {
+    dateParts[fromFormat[i]] = dateValues[i];
+  }
+
+  if (dateParts['YYYY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  } else if (dateParts['YY'] < 30) {
+    dateParts['YYYY'] = `20${dateParts['YY']}`;
+  } else if (dateParts['YY'] >= 30) {
+    dateParts['YYYY'] = `19${dateParts['YY']}`;
+  }
+
+  const newDateArr = [];
+  const newSeparator = toFormat[toFormat.length - 1];
```

Comment:
Similar to the issue with `fromFormat`, assuming the last element of `toFormat` is the separator might not be correct. Consider explicitly defining the separator if necessary.